### DISCUSSION TITLE
JK_B2A20S20P unable to locate MAC address or connect via BT

### DISCUSSION THREAD

**Discussion by austin202220** [syssi/esphome-jk-bms#232] [Q&A] [answered]
JK_B2A20S20P
Hardware: V11.XW
Software: V11.24H
Version: V4.10.1

I am lost... trying to connect via bluetooth. - theres no need to connect the uart rx and tx pins to the esp32 when connecting via bluetooth right??

When I review the logs I just get the following over and over in yellow text:

[17:14:42][W][jk_bms_ble:362]: [F0:00:FF:C2:04:51] Not connected

I never get any other devices listed and have not been able to locate the mac address of the BMS. I can connect to it on my iPhone no problem.

<img width="846" alt="Screenshot 2023-01-09 at 5 50 35 PM" src="https://user-images.githubusercontent.com/97492072/211430946-3b9a4313-0ea3-4dde-9edc-830ed3ff92c1.png">
<img width="854" alt="Screenshot 2023-01-09 at 5 15 56 PM" src="https://user-images.githubusercontent.com/97492072/211427270-4a4e188b-5eb4-439e-ae88-b90134343a86.png">

<img width="400" src="https://user-images.githubusercontent.com/97492072/211431627-e335487d-13ee-40dc-9ff1-a5a3a10f30a7.jpg">
<img width="904" alt="Screenshot 2023-01-09 at 5 59 33 PM" src="https://user-images.githubusercontent.com/97492072/211431877-64c5a1b8-14db-4ec9-a0cc-3023f31ffb54.png">

**Comment by syssi** [comment]
If you talk using BLE to the BMS no cables between the BMS and the ESP32 are required. Please change the `log_level` from `INFO` to `DEBUG` and flash the ESP again. This time you should see a list of visible BLE devices on boot-up. If there are none the BLE stack of your ESP is probably damaged.

**Comment by austin202220** [comment, marked as answer]
Than you @syssi for your help and all you have done! Unfortunately I never was able to get it to work to find the mac address that way the other thread describes.

I ended up going to home assistant terminal and entering in the command "bluetoothctl devices" and eventually got it to show on there. 

I'm posting a screenshot just in case anyone else stumbles into this later on maybe it will offer a 2nd way to get it for them too. I spy the JK in the list

<img width="1183" alt="Screenshot 2023-01-09 at 11 56 56 PM" src="https://user-images.githubusercontent.com/97492072/211472943-92733582-0198-4d2d-9323-581663cd3b13.png">

**Reply by RondaYummy** [reply]
Does anyone know what is the beginning of MAC addresses for JK-BMS devices?

**Comment by austin202220** [comment]
Oh also, on this model I have a newer software version than what is listed in the supported list. So I can confirm that the JK_B2A20S20P with Software V11.24H is working great over bluetooth if you want to add it

**Reply by syssi** [reply]
Done: https://github.com/syssi/esphome-jk-bms/commit/75b27cb89e14b463a5624a57be3cdb88f5d8d290

**Reply by syssi** [reply]
@austin202220 Please use protocol version `JK02_32S` here and avoid the auto-detection of the new (JK02_32S) frames because it's unstable.

**Comment by kamahat** [comment]
I've used my Mobile phone with the app "BLE Scanner (connect & notify)" and i've been able to retrive the MAC address. It's 6 bloc Hexa value . Mine start With C8:47:xx:xx:xx:xx
